funsor/constant.py

```diff
@@ -0,0 +1,128 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from collections import OrderedDict
+
+from funsor.distribution import Distribution
+from funsor.tensor import Tensor
+from funsor.terms import (
+    Binary,
+    Funsor,
+    FunsorMeta,
+    Number,
+    Unary,
+    Variable,
+    eager,
+    to_data,
+    to_funsor,
+)
+from funsor.torch.provenance import ProvenanceTensor
+
+from .ops import BinaryOp, UnaryOp
+
+
+class ConstantMeta(FunsorMeta):
+    """
+    Wrapper to convert ``const_inputs`` to a tuple.
+    """
+
+    def __call__(cls, const_inputs, arg):
+        if isinstance(const_inputs, dict):
+            const_inputs = tuple(const_inputs.items())
+
+        return super(ConstantMeta, cls).__call__(const_inputs, arg)
+
+
+class Constant(Funsor, metaclass=ConstantMeta):
```

Comment:
Interesting! It would probably be easiest for us to go over this PR and https://github.com/pyro-ppl/pyro/pull/2893 over Zoom, but one thing that would help me beforehand is if you could add a docstring here explaining how `Constant` behaves differently from `Delta` wrt `Reduce/Contraction/Integrate`